### PR TITLE
Adds palette to google charts

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @jbrundage @gururx 

Please review

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
I'm wondering if this should just be handled in common.js, and in each widget call Common.prototype.update.apply(this, arguments); before anything else.

this is for google widgets in: /src/google/

**Comment by GordonSmith** [comment]
As far as I can see the "_palette" code is already in place and working - what is it your trying to "fix"?

**Comment by mlzummo** [comment]
if i don't have this code, i can't switch palette's like so: 

https://securedev02.accurint.com/zummma01/viz5/widget_google_bar.php
click on disocver params and switch the color palette. I have it commented out in Bar.js the others are enabled.

**Comment by GordonSmith** [comment]
"I'm wondering if this should just be handled in common.js, and in each widget call Common.prototype.update.apply(this, arguments); before anything else." - Yes I would use this approach...

File: widgets/src/google/Common.js [hunk 1]
```diff
@@ -21,6 +21,12 @@
     Common.prototype = Object.create(HTMLWidget.prototype);
     Common.prototype.implements(I2DChart.prototype);
 
+    Common.prototype.publish("paletteID", "default", "set", "Palette ID", Common.prototype._palette.switch());
+
+    Common.prototype.update = function(domNode, element) {
+        this._palette = this._palette.switch(this._paletteID);
```
Comment (GordonSmith) [inline widgets/src/google/Common.js]: This should call up to the base class also...

File: widgets/src/google/Line.js [hunk 2]
```diff
@@ -16,6 +16,7 @@
     Line.prototype = Object.create(Common.prototype);
 
     Line.prototype.enter = function (domNode, element) {
+
```
Comment (GordonSmith) [inline widgets/src/google/Line.js]: Stray line

File: widgets/src/google/Pie.js [hunk 3]
```diff
@@ -38,6 +38,7 @@
     }
 
     Pie.prototype.enter = function (domNode, element) {
+
```
Comment (GordonSmith) [inline widgets/src/google/Pie.js]: Stray line

**Comment by mlzummo** [comment]
fixed

File: widgets/src/google/Common.js (hunk 1 above)
Comment (mlzummo) [inline widgets/src/google/Common.js]: fixed

**Comment by GordonSmith** [comment]
Looks good.

**Pull request merged**